### PR TITLE
Use Fusion Rendering

### PR DESCRIPTION
**Pull request by nilsjung**
This MR replaces the fluid template rendering with the fusion rendering aproach.

### CODE REVIEW

File: Resources/Private/Fusion/Components/Organisms/Navbar.fusion [hunk 1]
```diff
@@ -0,0 +1,63 @@
+prototype(Networkteam.Neos.Mockup:Components.Organisms.Navbar) < prototype(Neos.Fusion:Component) {
+    items = null
```
Comment (suffle) [inline Resources/Private/Fusion/Components/Organisms/Navbar.fusion]:
```suggestion
    items = ${[]}
```

File: Resources/Private/Fusion/Components/Organisms/Navbar.fusion [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                        <Networkteam.Neos.Mockup:Components.Organisms.Navbar.ItemList items={props.items} />
+                    </ul>
+                </div>
+            </div>
+        </div>
+    `
+}
+
+prototype(Networkteam.Neos.Mockup:Components.Organisms.Navbar.ItemList) < prototype(Neos.Fusion:Component) {
+    items = null
```
Comment (suffle) [inline Resources/Private/Fusion/Components/Organisms/Navbar.fusion]:
```suggestion
    items = ${[]}
```

File: Resources/Private/Fusion/Components/Organisms/Navbar.fusion [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            />
+        </Neos.Fusion:Loop>
+    `
+}
+
+prototype(Networkteam.Neos.Mockup:Components.Organisms.Navbar.ItemList.Item) < prototype(Neos.Fusion:Component) {
+    label = ''
+    state = ''
+    link = ''
+    subitems = ''
```
Comment (suffle) [inline Resources/Private/Fusion/Components/Organisms/Navbar.fusion]:
```suggestion
    subitems = ${[]}
```

File: Resources/Private/Fusion/Components/Molecules/Header.fusion [hunk 4]
```diff
@@ -0,0 +1,7 @@
+prototype(Networkteam.Neos.Mockup:Components.Molecules.Header) < prototype(Neos.Fusion:Component) {
+    mainNavigation = null
```
Comment (suffle) [inline Resources/Private/Fusion/Components/Molecules/Header.fusion]:
```suggestion
    mainNavigation = ''
```

File: Resources/Private/Fusion/Components/Atoms/Placeholder.fusion [hunk 5]
```diff
@@ -0,0 +1,9 @@
+prototype(Networkteam.Neos.Mockup:Components.Atoms.Placeholder) <  prototype(Neos.Fusion:Component) {
+    attributes = ''
+    placeholder = null
```
Comment (suffle) [inline Resources/Private/Fusion/Components/Atoms/Placeholder.fusion]:
```suggestion
    placeholder = ''
```

File: Resources/Private/Fusion/Content/Placeholder.fusion [hunk 6]
```diff
@@ -1,11 +1,16 @@
-prototype(Networkteam.Neos.Mockup:Placeholder) <  prototype(Neos.Neos:Content) {
-    attributes {
-        class = Neos.Fusion:RawArray {
-            base = 'neos-mockup-placeholder'
-            @process.nodeType >
+prototype(Networkteam.Neos.Mockup:Placeholder) <  prototype(Neos.Neos:ContentComponent) {
+    @context.containerHeight = ${q(node).property('height')}
+
+    renderer = Networkteam.Neos.Mockup:Components.Atoms.Placeholder {
+        attributes = Neos.Fusion:DataStructure {
```
Comment (suffle) [inline Resources/Private/Fusion/Content/Placeholder.fusion]: a lot of what is done here can also be done in the component itself, especially the style attribute should not be of any concern to the  mapping component

File: Resources/Private/Fusion/Documents/Document.Menu.fusion [hunk 7]
```diff
@@ -0,0 +1,70 @@
+prototype(Networkteam.Neos.Mockup:Document.Menu) < prototype(Neos.Fusion:Value) {
```
Comment (suffle) [inline Resources/Private/Fusion/Documents/Document.Menu.fusion]: If you use a `Neos.Fusion:Component` you dont have to push the api-values to the context manually

File: Resources/Private/Fusion/Documents/Document.Menu.fusion [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+prototype(Networkteam.Neos.Mockup:Document.Menu.Item.State) < prototype(Neos.Fusion:Case) {
+    current {
+        condition = ${item == documentNode}
+        renderer = 'active'
+    }
+
+    active {
+        condition = ${Array.indexOf(rootline, item) != -1}
```
Comment (suffle) [inline Resources/Private/Fusion/Documents/Document.Menu.fusion]: Using a context variable, that is defined several steps above is always a bit hard to understand, maybe we should use a `Neos.Fusion:Component` here a s well and set the API explicitly

**Review by suffle — CHANGES_REQUESTED** [review]

**Comment by nilsjung** [comment]
I adjusted the menu to use `Neos.Fusion:Component` instead of `Neos.Fusion:Value` and changed the type of the default values.

File: Resources/Private/Fusion/Documents/Document.Menu.fusion [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+prototype(Networkteam.Neos.Mockup:Document.Menu.Item) < prototype(Neos.Fusion:Component) {
+    item = null
+    menuLevel = 0
+    maximumLevels = 4
+
+    renderer = Neos.Fusion:DataStructure {
+        identifier = ${props.item.identifier}
+        label = ${q(props.item).property('title')}
+
+
```
Comment (suffle) [inline Resources/Private/Fusion/Documents/Document.Menu.fusion]: Remove unnecessary new line

File: Resources/Private/Fusion/Documents/Document.Menu.fusion [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+        link = Neos.Neos:NodeUri {
+            node = ${props.item}
+        }
+
+
```
Comment (suffle) [inline Resources/Private/Fusion/Documents/Document.Menu.fusion]: Remove unnecessary new line

**Review by suffle — CHANGES_REQUESTED** [review]

**Review by suffle — APPROVED** [review]

**Pull request merged**